### ISSUE TITLE
Setters and getters can't be defined in GDScript due to a syntax error

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#42177]
**Godot version:** Git a332e2f5b

**OS/device including version:** Fedora 31, GeForce GTX 1080 (NVIDIA 450.66)

**Issue description:**

I get a syntax error when trying to define a property with a setter method (regardless if there's a getter method before it or not).

**Steps to reproduce:**

Write the following code in a script (identical to the one in the MRP):

```gdscript
extends Node2D

var some_property: int
	get: # <-- ERROR: Unexpected "Indent" in class body.
		print("Getting some_property")
		return some_property
	set(value):
		print("Setting some_property to %s" % value)
		some_property = value


@export var some_exported_property: int
	get:
		print("Getting some_exported_property")
		return some_exported_property
	set(value):
		print("Setting some_exported_property to %s" % value)
		some_exported_property = value
```

**Minimal reproduction project:** [export-setget-bug.zip](https://github.com/godotengine/godot/files/5247945/export-setget-bug.zip)

**Comment by ThakeeNathees** [comment]
I've fixed it but not sure about the syntax with the type, should it have a colon at the end after the type name or not ? (I prefer it with a colon at the end)
```gdscript
var prop: int
	get:
```
```gdscript
var prop: int:
	get:
```
@Calinou @vnen

**Comment by ThakeeNathees** [comment]
using the property in the getter or setter cause a crash (the above code in the issue). it should throw a parser error.
```gdscript
var prop: int:
	get:
		return prop
	set(value):
		prop = value
```

instead it should be used with another internal variable
```gdscript
var _prop:int
var prop: int:
	get:
		return _prop
	set(value):
		_prop = value
```

**Comment by KoBeWi** [comment]
Seems fixed in fe93f62
Add colon after `int` in the original code and it will work (except it crashes, but syntax is ok at least)
